# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: xpcom/threads/nsThread.cpp
# Commit: 31f72cc99309
# Full Hash: 31f72cc99309dddfb6cb9aa7a4862dacfe5be45f
# Author: Karl Tomlinson <karlt+@karlt.net>
# Date: 2019-12-13 16:53:24
# Description:
#   Bug 1603153 throw an exception from nsThread::GetPRThread() when there is no PRThread r=froydnj
#   
#   as is done in LazyIdleThread::GetPRThread().
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D56826
# ==============================================================================

diff -r 309384fdbe94 -r 31f72cc99309 xpcom/threads/nsThread.cpp
--- a/xpcom/threads/nsThread.cpp	Fri Dec 13 07:31:54 2019 +0000
+++ b/xpcom/threads/nsThread.cpp	Thu Dec 12 21:53:58 2019 +0000
@@ -776,8 +776,9 @@
 
 NS_IMETHODIMP
 nsThread::GetPRThread(PRThread** aResult) {
-  *aResult = mThread;
-  return NS_OK;
+  PRThread* thread = mThread;  // atomic load
+  *aResult = thread;
+  return thread ? NS_OK : NS_ERROR_NOT_AVAILABLE;
 }
 
 NS_IMETHODIMP
